R/dtree_cross_validation.R

```diff
@@ -0,0 +1,531 @@
+            }
+            AlteryxMessage2(warningMessage1)
+            AlteryxMessage2(warningMessage2)
+            AlteryxMessage2(errorMessage)
+          }
+        }
+      }
+    }
+  }
+}
+
+
+#' Create the list of cross-validation folds
+#'
+#' @param data the data.frame used to create the models
+#' @param config a list of configuration information
+#' @return list of record ID's. Each element is the record ID's of the folds for a given trial.
+#' @import TunePareto
+createFolds <- function(data, config) {
+  target <- data[, 1]
+  if (config$set_seed_cv) {
+    set.seed(config$cv_seed)
+  }
+  foldList <- TunePareto::generateCVRuns(labels = target, ntimes = config$numberTrials, nfold = config$numberFolds, stratified = config$stratified)
+  checkFactorVars(data = data, folds = foldList, config = config)
+  return(foldList)
+}
+
+
+# In the 2-class classification case, get the positive class. Otherwise, do nothing.
+
+getPosClass <- function(config, yVar) {
+
+  #Use the function from the Model Comparison tool to get/set positive class:
+  setPositiveClass <- function(tar_lev) {
+    # Set the postive class for two-class classification.
+    # The setPositiveClass function is only triggered if the user leaves the
+    # question on positive class (target level) blank.
+    #   1) if there's "yes/Yes/YES ..." in the target variable, then use "yes/Yes/YES"
+    #   2) if there's "true/True/TRUE..." in the target variable, then use "true/True/TRUE"
+    #   3) otherwise: use the less common class
+    #
+    # Parameters:
+    #   tar_lev: a vector of string
+    #            the levels of the target variable.
+    #
+    # Returns:
+    #   no_name: a string, the name of the positive class.
+
+    yes_id <- match("yes", tolower(tar_lev))
+    true_id <- match("true", tolower(tar_lev))
+    if (!is.na(yes_id)) {
+      return (tar_lev[yes_id])
+    } else if (!is.na(true_id)) {
+      return (tar_lev[true_id])
+    } else {
+      first_class <- tar_lev[1]
+      second_class <- tar_lev[which(tar_lev != first_class)[1]]
+      if ((length(which(tar_lev) == first_class)) > (length(which(tar_lev) == second_class))) {
+        #First_class is larger, so second_class is the positive class
+        return (second_class)
+      } else {
+        return (first_class)
+      }
+    }
+  }
+  return(setPositiveClass(yVar))
+}
+
+
+# Given a model, a dataset and index of test cases, return actual and response
+#' @import C50
+#' @import rpart
+getActualandResponse <- function(model, data, testIndices, extras, mid){
+  trainingData <- data[-testIndices,]
+  testData <- data[testIndices,]
+  testData <- matchLevels(testData, getXlevels(model))
+  currentYvar <- getYVar(model)
+  currentModel <- update(model, data = trainingData)
+  pred <- AlteryxPredictive::scoreModel(currentModel, new.data = testData)
+  actual <- (extras$yVar)[testIndices]
+  recordID <- (data[testIndices,])$recordID
+  if (config$classification) {
```

Comment:
You are using `config`, but it is not an argument to the function. I presume you will need to (1) update its signature to accept `config` as an argument and (2) pass `config` to it where you call it.